tools/release_model.lua

```diff
@@ -39,7 +46,15 @@ local function releaseModule(object, tensorCache)
   if object.release then
     object:release()
   end
-  object:float(tensorCache)
+  object:apply(function (m)
+    for k, v in pairs(m) do
+      if torch.isTensor(v) and v:dim()>0 then
+        v:mul(1000)
+        m[k] = v:short()
```

Comment:
Note that this cancels weights sharing. This concerns for example word embeddings in bidirectional encoders.

See https://github.com/OpenNMT/OpenNMT/commit/de97aa259c01f15769a4550637bbd807b852d490.